common/proto-conv/src/config_from_to_protobuf_impl.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_configs::S3StorageConfig;
+use common_protos::pb;
+
+use crate::FromToProto;
+use crate::Incompatible;
+
+impl FromToProto<pb::S3StorageConfig> for S3StorageConfig {
+    fn from_pb(p: pb::S3StorageConfig) -> Result<Self, Incompatible>
+    where Self: Sized {
+        Ok(Self {
+            region: p.region,
+            endpoint_url: p.endpoint_url,
+            access_key_id: p.access_key_id,
+            secret_access_key: p.secret_access_key,
+            bucket: p.bucket,
+            root: p.root,
+            master_key: p.master_key,
+        })
+    }
+
+    fn to_pb(&self) -> Result<pb::S3StorageConfig, Incompatible> {
+        Ok(pb::S3StorageConfig {
+            region: self.region.clone(),
+            endpoint_url: self.endpoint_url.clone(),
+            access_key_id: self.access_key_id.clone(),
```

Comment:
@drmingdrmer Is it safe to store ak/sk in meta directly? Are there some consideration here?

Reply:
hmm... metasrv would be a nice place to store it if with some strong encryption. 🤔